src/tasks/trade-history-task.ts

```diff
@@ -0,0 +1,52 @@
+import { getRepository } from 'typeorm'
+import { queryTrades } from '../utils/subgraph'
+import { setLastTaskRun } from '../models/task.model'
+import { TaskType } from '../entity/Task'
+import { tradeSearchCron } from '../crontab'
+import { Trade } from '../entity'
+import dayjs from 'dayjs'
+
+let lastTimestamp: dayjs.Dayjs | null = null
+
+export const getTradesTask = async (): Promise<void> => {
+  const currentTimestamp = dayjs()
+  tradeSearchCron.stop()
+
+  try {
+    // If it's the first execution, set lastTimestamp and return
+    if (lastTimestamp === null) {
+      lastTimestamp = currentTimestamp.subtract(24, 'hours')
+      console.log('getTradesTask() executed for the first time.')
+    } else {
```

Comment:
By using `else` here you are never running task for the first time.
Set lastTimestamp to 24hours ago for the first run, and then just continue with code normally.